tools/NumGrid/mk_grid.py

```diff
@@ -0,0 +1,160 @@
+        element.append(l[0])
+        x_coordinates_bohr.append(float(l[2]))
+        y_coordinates_bohr.append(float(l[3]))
+        z_coordinates_bohr.append(float(l[4]))
+    
+        # print("idxb = ", idxb)
+
+        idxk = 0
+        for j in range(idxb):
+            # print("j", j)
+            for i in range(block[j]):
+                # print("block[j]", block[j])
+                l = f.readline().strip().split()
+                # print(l)
+                nfunc = int(l[0]) 
+                # print("i", i)
+                # print("nfunc =", nfunc)
+                for k in range(nfunc):
+                    l = f.readline().strip().split()
+                    alpha.append(float(l[0]))
+                    # print("k", k)
+                    idxk = idxk + 1
+            alpha_min_tmp.append(alpha[idxk-1])
+
+#       print("alpa", alpha)
+        alpha_max.append(alpha[0])
+        alpha_min.append(alpha_min_tmp[:])
+        # clean up for new element
+        idxb = 0
+        block[:] = []
+        alpha[:] = []
+        alpha_min_tmp[:] = []
+
+# print("alpa_max", alpha_max)
+# print("alpa_min", alpha_min)
+# print("")
+# print("alpha_min for center 1", alpha_min[0])
+# print("")
+# print("alpha_min for center 7", alpha_min[6])
+# print(x_coordinates_bohr, y_coordinates_bohr, z_coordinates_bohr)
+# print("max L", max_l_quantum_numbers) 
+fout = "coord.au"
+f1 = open(fout,"w")
+for n in range(num_centers):
+    f1.write(str(x_coordinates_bohr[n])+" "+str(y_coordinates_bohr[n])+" "+str(z_coordinates_bohr[n])+"\n")
+
+f1.close()
+
+# now calculate the grid after all input has been extracted from MOL
+
+fout_grid = "gridfile.grd"
+fout_weights = "grid_w.grd" 
+f1 = open(fout_grid, "w")
+f2 = open(fout_weights, "w")
+
+for n in range(num_centers):
+    print("n", n)
+    print("proton_charges[n]", proton_charges[n])
+    print("alpha_max[n]", alpha_max[n])
+    print("max_l_quantum_numbers[n]", max_l_quantum_numbers[n])
+    print("alpha_min[n]", alpha_min[n])
+
+    context = numgrid.new_atom_grid(radial_precision, 
+            min_num_angular_points, max_num_angular_points, 
+            proton_charges[n], alpha_max[n], max_l_quantum_numbers[n], 
+            alpha_min[n] )
+
+    num_points = numgrid.get_num_grid_points(context)
+    print("number of points", num_points)
+    # generate an atomic grid in the molecular environment
+    x, y, z, w = numgrid.get_grid(context, num_centers, n, 
+            x_coordinates_bohr, y_coordinates_bohr, 
+            z_coordinates_bohr, proton_charges)
+
+    # np.save('grid_xyz.npy', x, y, z) 
+    # np.save('grid_w.npy', w)
+
+    for k in range(num_points):
+        f1.write(str(x[k]) + " " + str(y[k]) + " " + str(z[k]) + "\n")
+        f2.write(str(w[k]) + "\n")
+
+    # num_radial_points = numgrid.get_num_radial_grid_points(context)
+    # print("number of radial points", num_radial_points)
+
+    # generate an isolated radial grid
+    # r, w = numgrid.get_radial_grid(context)
```

Comment:
we should also remove all the out-commented unused code

Reply:
Yes, I know. I tested if it works. If we keep it it would be easy to adapt the script, but we can also remove it.

Reply:
Ok, cleaned up.